File: frontend/src/types/databaseAdmin.ts
/**
 * 数据库管理相关类型定义
 */

export interface AdminPermission {
  has_database_admin: boolean
  has_user_admin: boolean
  has_permission_admin: boolean
}

export interface DatabaseInfo {
  name: string
  charset?: string
  collation?: string
  encoding?: string
  size?: number
  table_count?: number
  created_at?: string
}

export interface CreateDatabaseRequest {
  name: string
  charset?: string
  collation?: string
  encoding?: string
  lc_collate?: string
  lc_ctype?: string
}

export interface RenameDatabaseRequest {
  new_name: string
}

File: backend/internal/models/database_admin.go
package models

// AdminPermissionResponse 管理权限响应
type AdminPermissionResponse struct {
	HasDatabaseAdmin  bool `json:"has_database_admin"`
	HasUserAdmin      bool `json:"has_user_admin"`
	HasPermissionAdmin bool `json:"has_permission_admin"`
}

// CreateDatabaseRequest 创建数据库请求
type CreateDatabaseRequest struct {
	Name      string `json:"name" binding:"required"`
	Charset   string `json:"charset,omitempty"`   // MySQL
	Collation string `json:"collation,omitempty"` // MySQL
	Encoding  string `json:"encoding,omitempty"`  // PostgreSQL
	LcCollate string `json:"lc_collate,omitempty"` // PostgreSQL
	LcCtype   string `json:"lc_ctype,omitempty"`   // PostgreSQL
}

// RenameDatabaseRequest 重命名数据库请求
type RenameDatabaseRequest struct {
	NewName string `json:"new_name" binding:"required"`
}
